### PR TITLE
Migrate the interface to the latest version of Bootstrap

### PR DESCRIPTION
**Pull request by nikhilgupta1211**
Updated the twitter-bootstrap-rails gem to 3.2.2 version
Generated the less files needed for the updated version

Updated the layout of user guide page with bootstrap 3 classes

When updating the twitter-bootstrap-rails gem I had to restrict the rake from getting updated to version later than 11.0 as Rake 11.0.1 removes the last_comment method which rspec-core uses.

### CODE REVIEW

File: Gemfile.lock [hunk 1]
```diff
@@ -188,7 +188,7 @@ GEM
       activesupport (= 4.1.10)
       rake (>= 0.8.7)
       thor (>= 0.18.1, < 2.0)
-    rake (12.0.0)
+    rake (10.5.0)
```
Comment (bgeuken) [inline Gemfile.lock]: What's the reason for downgrading rake?

File: spec/support/common_helpers.rb [hunk 2]
```diff
@@ -150,10 +150,10 @@ def transition(machine, state_event, user)
   # Closes a modal dialog and waits until it's really closed
   #
   def close_modal_dialog
-    within(".modal") do
+    within(".modal-diaglog") do
       click_link "Back"
     end
-    page.should_not have_css(".modal-backdrop", :wait => 15)
+    page.should_not have_css(".modal-backdrop", :wait => 30)
```
Comment (bgeuken) [inline spec/support/common_helpers.rb]: Why did you change it to 30?

Comment (nikhilgupta1211) [inline spec/support/common_helpers.rb]: Actually I was trying to fix the test but the problem is not here so I will revert this

File: Gemfile.lock (hunk 1 above)
Comment (nikhilgupta1211) [inline Gemfile.lock]: Rake 11.0.1 removes the last_comment method which rspec-core uses. So we have to either add this method in the rake file manually when using updated version or have to use the versions before 11.0

File: spec/support/common_helpers.rb [hunk 3]
```diff
@@ -150,10 +150,10 @@ def transition(machine, state_event, user)
   # Closes a modal dialog and waits until it's really closed
   #
   def close_modal_dialog
-    within(".modal-diaglog") do
+    within(".modal") do
       click_link "Back"
     end
-    page.should_not have_css(".modal-backdrop", :wait => 30)
+    page.should_not have_css(".modal-backdrop", :wait => 15)
```
Comment (bgeuken) [inline spec/support/common_helpers.rb]: Since this got introduced in the same PR, it would be nice to remove it completely from the commit history. In other words, to squash it into 372efc494ffc0a6aba55d75fad1bd578277775db.

That's a bit advanced and might go wrong the first time. So I would recommend to make a copy of your branch first.

The best way to revert this change might be:

```

git rebase -i 372efc494ffc0a6aba55d75fad1bd578277775db^
# Change the first line (the one for 372efc494ffc) from pick xyz to edit xyz
# Once you closed the editor checked out the state of 372efc494ffc
# Edit spec/support/common_helpers.rb
git add spec/support/common_helpers.rb
git rebase --continue
# solve any merge conflicts
```

[gitread.com](http://gitready.com/advanced/2009/02/10/squashing-commits-with-rebase.html) also explains this nicely. They have a nice overview of and explanations for various git commands.

Comment (nikhilgupta1211) [inline spec/support/common_helpers.rb]: what if I squash the fix tests failing commit into the updated the class commit ?

Comment (bgeuken) [inline spec/support/common_helpers.rb]: If all changes that get introduced an this commit, where either introduced in 372efc or belong to that commit, I would do that.

Comment (nikhilgupta1211) [inline spec/support/common_helpers.rb]: Sorry I didn't get you, the commit 500843e6cabc26e3f604cf21fe649704a130ccfe contains only some of the corrections for the commit 372efc494ffc0a6aba55d75fad1bd578277775db. So should I squash them?

**Comment by bgeuken** [comment]
Are you following the bootstrap migration guide[1][2]?

eg. when you go to http://localhost:3000/events you currently see a big link 'Travel support'. Instead it should be a small button.

The class for that element should be ``` class="btn btn-default btn-xs" ``` instead of ``` class="btn btn-small" ```. That's [this](https://github.com/openSUSE/travel-support-program/pull/47/files#diff-3a346dc2129e48d0544d1c9f612821aeL38) line.

[1] https://www.bootply.com/bootstrap-3-migration-guide
[2] http://getbootstrap.com/migration/

**Comment by nikhilgupta1211** [comment]
Yes I have been following that guide. Sorry I see I have missed some changes.

File: app/views/events/index.html.haml [hunk 4]
```diff
@@ -2,18 +2,19 @@
 = content_for :page_header do
   %h1= t('.title', :default => model_class.model_name.human(count: 2))
 
-.navbar
-  .navbar-inner
-    = link_to t('.new', :default => t("helpers.links.new")), new_resource_path, :class => 'btn btn-primary', :data => { :confirm => t(:new_event_warning) }
+%nav.navbar.navbar-default
+  .container-fluid
```
Comment (bgeuken) [inline app/views/events/index.html.haml]: @nikhilgupta1211 The .container-fluid causes the 'New' button to be aligned differently (some pixels further to the right). Also the form differs quite a bit from what it look like before.

Latter one seems to be caused by replacing 'input-small' with 'form-control'. The [bootstrap docs](http://getbootstrap.com/migration/#dropped) recommend to "Use .form-control and the [grid system](http://getbootstrap.com/css/#grid) instead.".
You might have to apply this here.

**Comment by bgeuken** [comment]
@nikhilgupta1211 Also the sidebar navigation lists are not formatted correctly. There is to much spacing and the 'Options' header looks wrong.

It might be that our bootstrap overwrites have to be adopted after the bootstrap update.

The same might be true for the appearance of buttons and the top navigation bar.

File: app/assets/javascripts/bootstrap.js.coffee [hunk 5]
```diff
@@ -1,11 +1,3 @@
-#
-#= require twitter/bootstrap
-#= require datetimepicker/bootstrap-datetimepicker
-#= require bootstrap-fileupload
-#= require bootstrap-multiselect
```
Comment (bgeuken) [inline app/assets/javascripts/bootstrap.js.coffee]: What's actually the reason for removing those? You didn't mention that in the commit message

Comment (bgeuken) [inline app/assets/javascripts/bootstrap.js.coffee]: This gets added again in the next commit. It would be good to rebase such reverted changes into the previous commit.

File: app/views/events/index.html.haml [hunk 6]
```diff
@@ -2,18 +2,20 @@
...
 
+    .nav.navbar-right
+      = search_form_for @q, :builder => SimpleForm::FormBuilder, :class => "navbar-form" do |f|
+        - t_start = f.input_field :end_date_gteq, :as => :dpicker, class:'form-control input-sm inpt-w'
+        - t_end = f.input_field :start_date_lteq, :as => :dpicker, class:'form-control input-sm inpt-w'
+        - t_text = f.text_field :name_or_description_cont, class:'form-control input-sm inpt-w'
+        = raw(t(:events_search, :start => t_start, :end => t_end, :text => t_text))
+        = f.submit t(:search), :class => "btn btn-default"
+        = link_to t(:reset_search), collection_path(:q => {:end_date_gteq => '0'}), :class => "btn btn-default"
+
```
Comment (bgeuken) [inline app/views/events/index.html.haml]: You introduced some trailing whitespaces here, and at some other places. Please remove them.

File: app/assets/stylesheets/bootstrap_overrides.less [hunk 7]
```diff
@@ -33,16 +33,102 @@ body {
...
+  }
+
+  .nav-main{
+    height: 42px;
+  }
+}
 /* Set the fixed height of the footer here */
 #push-footer, #footer {
-  height: @baseFontSize*1.5;
+  height: 30px;
```
Comment (bgeuken) [inline app/assets/stylesheets/bootstrap_overrides.less]: @nikhilgupta1211 There actually still is a variable that defines the default for the font size. Why don't you use them?

And that's also true for some of the other variables we have been using.

Comment (nikhilgupta1211) [inline app/assets/stylesheets/bootstrap_overrides.less]: Because of the syntax change in less, in the versions above 2.4 you can use variables directly without needing ~hack but the file mixins.less in the twitter-bootstrap-rails uses the old syntax so it gives an error of unrecognized input. We are able to use it in the bento theme as we are defining the variables in bootstrap_variables.less in the bento directory.

Comment (bgeuken) [inline app/assets/stylesheets/bootstrap_overrides.less]: Actually @baseFontSize just got renamed. It's still mentioned in current version of [bootstrap documentation](http://getbootstrap.com/css/#type-headings)

Comment (nikhilgupta1211) [inline app/assets/stylesheets/bootstrap_overrides.less]: Yes changed it, thanks :)

File: app/views/layouts/application.html.haml [hunk 8]
```diff
@@ -13,35 +13,36 @@
 
   %body
     #full-page-wrap
-      .navbar.navbar-fluid-top
-        .navbar-inner
-          .container-fluid
-            %a.btn.btn-navbar(data-target=".nav-collapse" data-toggle="collapse")
```
Comment (bgeuken) [inline app/views/layouts/application.html.haml]: There is some code that depends on this html class. It's only used by the bento theme. If this isn't needed anymore we can drop the js code. Otherwise we have to fix it?

https://github.com/openSUSE/travel-support-program/blob/master/app/themes/bento/assets/javascripts/bento/global-navigation.js#L30

File: app/assets/stylesheets/bootstrap_overrides.less [hunk 9]
```diff
@@ -33,16 +33,102 @@ body {
...
+  }
+}
 /* Set the fixed height of the footer here */
 #push-footer, #footer {
-  height: @baseFontSize*1.5;
+  height: @font-size-base*1.5;
 }
 #footer {
-  background-color: @navbarBackground;
+  background-color:@navbar-default-bg;
```
Comment (bgeuken) [inline app/assets/stylesheets/bootstrap_overrides.less]: There is a whitespace missing;-)

File: app/assets/stylesheets/bootstrap_overrides.less (hunk 7 above)
Comment (bgeuken) [inline app/assets/stylesheets/bootstrap_overrides.less]: :+1:

File: app/assets/stylesheets/bootstrap_overrides.less [hunk 10]
```diff
@@ -33,16 +33,102 @@ body {
...
+
+.form-report{ 
+  .input-sm {
+    width: 220px;
+   }
+
+   label{
+    font-weight: normal;
+   } 
+}
```
Comment (bgeuken) [inline app/assets/stylesheets/bootstrap_overrides.less]: It seems to me that the overwrites for page-header and form-report don't have a big effect. The only thing I notice when removing them is that the forms on the event page and elsewhere move a bit up.

If this is the only thing they touch, I would remove them.

File: app/themes/bento/views/layouts/_navbar_top.html.haml [hunk 11]
```diff
@@ -1,6 +1,12 @@
...
-  %ul.nav#global-navigation
+.navbar-header
+  %button.navbar-toggle.collapsed.btn-navbar{"aria-expanded" => "false", "data-target" => "#bs-example-navbar-collapse-1", "data-toggle" => "collapse", :type => "button"}
+    %span.sr-only Toggle navigation
+    %span.icon-bar
+    %span.icon-bar
+    %span.icon-bar
+  =link_to image_tag("bento/header-logo.png"), root_path, :class => "brand navbar-brand"
+#bs-example-navbar-collapse-1.collapse.navbar-collapse
+  %ul.nav.navbar-nav#global-navigation
```
Comment (bgeuken) [inline app/themes/bento/views/layouts/_navbar_top.html.haml]: Seems that these change break the gecko icon in bento mode

File: app/assets/stylesheets/bootstrap_overrides.less (hunk 10 above)
Comment (nikhilgupta1211) [inline app/assets/stylesheets/bootstrap_overrides.less]: form-report provides the width to the input fields in the report view we can remove the page-header as it is only providing the margin
